File: lib-database/src/main/java/com/ustadmobile/core/db/dao/PersonLocationJoinDao.java
package com.ustadmobile.core.db.dao;

import com.ustadmobile.lib.database.annotation.UmDao;
import com.ustadmobile.lib.database.annotation.UmRepository;
import com.ustadmobile.lib.db.entities.PersonLocationJoin;
import com.ustadmobile.lib.db.sync.dao.SyncableDao;

@UmDao(inheritPermissionFrom = PersonDao.class,
inheritPermissionForeignKey = "personLocationPersonUid",
inheritPermissionJoinedPrimaryKey = "personUid")
@UmRepository
public abstract class PersonLocationJoinDao implements SyncableDao<PersonLocationJoin, PersonLocationJoinDao> {


}

File: lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/dao/PersonLocationJoinDao.kt
package com.ustadmobile.core.db.dao

import androidx.room.Dao
import com.ustadmobile.lib.database.annotation.UmDao
import com.ustadmobile.lib.database.annotation.UmRepository
import com.ustadmobile.lib.db.entities.PersonLocationJoin

@UmDao(inheritPermissionFrom = PersonDao::class, inheritPermissionForeignKey = "personLocationPersonUid", inheritPermissionJoinedPrimaryKey = "personUid")
@Dao
@UmRepository
abstract class PersonLocationJoinDao : SyncableDao<PersonLocationJoin, PersonLocationJoinDao>
